sw/device/lib/dif/BUILD

```diff
@@ -250,6 +258,8 @@ cc_library(
 cc_test(
     name = "spi_device_unittest",
     srcs = [
+        "autogen/dif_spi_device_autogen.c",
+        "autogen/dif_spi_device_autogen.h",
```

Comment:
Is there a reason you didn't add the autogen unit tests here? (Applies throughout, since after I closed out #8409, and autogen'd all `dif_<ip>_init()` there should be no empty autogen unit test files now.)